rust/connlib/tunnel/src/dns.rs

```diff
@@ -220,8 +252,11 @@ fn build_response(original_pkt: IpPacket<'_>, mut dns_answer: Vec<u8>) -> IpPack
     let response_len = dns_answer.len();
     let original_dgm = original_pkt.unwrap_as_udp();
     let hdr_len = original_pkt.packet_size() - original_dgm.payload().len();
-    let mut res_buf = Vec::with_capacity(hdr_len + response_len);
+    let mut res_buf = Vec::with_capacity(hdr_len + response_len + 20);
 
+    // TODO: this is some weirdness due to how MutableIpPacket is implemented
+    // we need an extra 20 bytes padding.
+    res_buf.extend_from_slice(&[0; 20]);
```

Comment:
I've always found this function to be very odd. Why are operating on byte buffers? Can we use `ip_packet::make` here instead now?